src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -3074,8 +3076,9 @@ public object VisitPipeline(PipelineAst pipelineAst)
                 Expression invokePipe = Expression.Call(
                     CachedReflectionInfo.PipelineOps_InvokePipeline,
                     input.Cast(typeof(object)),
-                    firstCommandExpr != null ? ExpressionCache.FalseConstant : ExpressionCache.TrueConstant,
-                    Expression.NewArrayInit(typeof(CommandParameterInternal[]), pipelineExprs),
+                    firstCommandExpr != null && !pipelineAst.BackgroundProcess ? ExpressionCache.FalseConstant : ExpressionCache.TrueConstant,
+                    Expression.NewArrayInit(typeof(CommandParameterInternal[]), 
```

Comment:
After looking more closely, I wonder if it makes sense to avoid basically this whole method (`Compiler.VisitPipeline`) and generate code to call `PipelineOps.InvokePipeline` directly.

Of particular concern - the pipelineExprs are ignored - so they shouldn't be generated in the first place. You never know what sort of side effects generating those expressions might have, it's better to avoid that entirely.